framework/transport/ieee1905_transport/ieee1905_transport_packet_processing.cpp

```diff
@@ -271,29 +271,12 @@ bool Ieee1905Transport::fragment_and_send_packet_to_network_interface(unsigned i
 
     // reuse most of the original packet metadata for all fragments (except for the payload iov)
     Packet fragment_packet = packet;
-
-    uint8_t buf[kIeee1905FragmentationThreashold];
+    uint8_t buf[kIeee1905FragmentationThreashold + sizeof(Tlv)] = {0}; // fragment size + TLV end of message size
 
     // copy the IEEE1905 header (will be reused by all fragments)
     std::copy_n((uint8_t *)packet.payload.iov_base, sizeof(Ieee1905CmduHeader), buf);
     int remainingPacketLength = packet.payload.iov_len - sizeof(Ieee1905CmduHeader);
 
-// packed TLV struct to help in the parsing of the packet
```

Comment:
I prefer to have such "moving code around" changes (that make no functional difference) in a separate commit. That makes it easier to spot the risky changes (because the "moving code around" carries almost no risk).

If part of the "moving code around" is changing something (because otherwise things break, e.g. because you're no longer in the same `using` context), then this should be mentioned explicitly in its commit message.

Reply:
Agree, but in this case there is an actual functional difference - I only needed the Tlv for using sizeof(Tlv) instead of using hard coded number for the size, so this forced me to move it to the header (which is where it should have been in the first place).

Reply:
> I only needed the Tlv for using sizeof(Tlv)

That's exactly what I mean. There is a reason for moving the code around, but moving the code around itself makes no functional difference. Normally we want to keep stuff that logically belongs together in a single commit. However, moving code around makes for big diffs, which is hard to review: there is more change to review, and it's hard to see if (as part of the move) anything actually changed. Therefore, I like to have the "moving code around" part as a separate commit - a commit that yields no functional change, but that is an enabler for a later commit.

Clearly, this is not an essential comment, and splitting the commit now is pointless. It's just for future reference that I mention this.